service/grails-app/services/org/olf/rs/PatronNoticeService.groovy

```diff
@@ -0,0 +1,132 @@
+import org.olf.rs.EventPublicationService
+import org.olf.rs.NoticePolicyNotice
+
+public class PatronNoticeService {
+
+  EmailService emailService
+  EventConsumerService eventConsumerService
+  EventPublicationService eventPublicationService
+  OkapiClient okapiClient
+  GrailsApplication grailsApplication
+  private KafkaConsumer consumer = null
+
+  @javax.annotation.PostConstruct
+  public void init() {
+    // We need another consumer because EventConsumerService consumes all topics
+    // as they come in whereas the notice triggers need to queue until they can
+    // be consumed under the aegis of a timer request
+    log.debug("Configuring event consumer for patron notice service")
+    Properties props = new Properties()
+    try {
+      grailsApplication.config.events.consumer.toProperties().each { final String key, final String value ->
+        // Directly access each entry to cause lookup from env
+        String prop = grailsApplication.config.getProperty("events.consumer.${key}")
+        log.debug("Configuring event consumer for patron notice service :: key:${key} value:${value} prop:${prop}");
+        props.setProperty(key, prop)
+      }
+      log.debug("Configure consumer ${props}")
+    }
+    catch ( Exception e ) {
+      log.error("Problem assembling props for consume",e);
+    }
+
+    consumer = new KafkaConsumer(props)
+    log.debug("PatronNoticeService::init() returning");
+  }
+
+  @javax.annotation.PreDestroy
+  private void cleanUp() throws Exception {
+    log.info("PatronNoticeService::cleanUp");
+    consumer.wakeup();
+  }
+
+  public void triggerNotices(PatronRequest pr, String trigger) {
+    if (!(pr?.patronEmail instanceof String)) {
+      log.error("Notice triggered for request ${pr.hrid} with no email for patron");
+      return;
+    }
+    log.debug("triggerNotices(${pr.patronEmail}, ${trigger})")
+    String tenant = Tenants.currentId()
+    eventPublicationService.publishAsJSON(
+      "${tenant}_PatronNoticeEvents",
+      null,
+      [
+        event:'TriggerNotices_ind',
+        tenant: tenant,
+        payload:[
+          trigger: trigger,
+          email: pr.patronEmail,
+          values: [
+            user: [
+              id: pr?.patronReference,
+              firstName: pr?.patronGivenName,
+              lastName: pr?.patronSurname,
+            ],
+            request: [
+              id: pr.hrid,
+              // pickupLocation: pr?.pickupLocation,
+              // neededBy: pr?.neededBy
+            ],
+            item: [
+              title: pr?.title
+            ]
+          ]
+        ]
+      ]
+    );
+  }
+
+  public void processQueue() {
+    log.debug("Processing patron notice queue")
+    def tenant_list = eventConsumerService.getTenantList()
+    if ( ( tenant_list == null ) || ( tenant_list.size() == 0 ) ) return
+    def topics = tenant_list.collect { "${it}_mod_rs_PatronNoticeEvents".toString() }
+    consumer.subscribe(topics)
```

Comment:
Subscribe every time we process in case new tenants are added ...perhaps I should also unsubscribe at the end of processing in case they are removed?